ui/lib/components/AutoRefreshButton/AutoRefreshButton.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { DownOutlined, LoadingOutlined, SyncOutlined } from '@ant-design/icons'
+import { Spin, Dropdown, Menu, Space } from 'antd'
+import { useSpring, animated } from 'react-spring'
+import { getValueFormat } from '@baurine/grafana-value-formats'
+import { useTranslation } from 'react-i18next'
+import { addTranslationResource } from '@lib/utils/i18n'
+import { useGetSet } from 'react-use'
+
+interface AutoRefreshButtonProps {
+  autoRefreshSecondsOptions: number[]
+  autoRefreshSeconds: number
+  onAutoRefreshSecondsChange: (v: number) => void
+  onRefresh: () => Promise<void>
```

Comment:
I think you actually don't need it to be async. `disabled` can be already used to express the information about whether the auto refresh ticking should continue. BTW your current implementation keep ticking when disabled = true, which is incorrect.